src/wald_wolfowitz.jl

```diff
@@ -0,0 +1,68 @@
+export WaldWolfowitzTest, RunsTest
+
+struct WaldWolfowitzTest{T<:Real} <: HypothesisTest
+    nabove::Int     # Number of points above median (or of value a)
+    nbelow::Int     # Number of points below median (or of value b)
+    nruns::Int      # Number of runs
+    μ::T            # Expected mean
+    σ::T            # Expected variance
+    z::T            # test z-statistic from Normal distribution
+end
+
+testname(::WaldWolfowitzTest) = "Wald-Wolfowitz Test"
+population_param_of_interest(x::WaldWolfowitzTest) = ("Number of runs", x.μ, x.nruns) # parameter of interest: name, value under h0, point estimate
+default_tail(::WaldWolfowitzTest) = :both
+pvalue(test::WaldWolfowitzTest; tail=:both) = pvalue(Normal(0.0, 1.0), test.z; tail=tail)
+
+
+function show_params(io::IO, x::WaldWolfowitzTest, ident="")
+    println(io, ident, "number of runs:  $(x.nruns)")
+    println(io, ident, "z-statistic:     $(x.z)")
+end
+
+"""
+    WaldWolfowitzTest(x::AbstractVector{<:Bool})
+    WaldWolfowitzTest(x::AbstractVector{<:Real})
+
+Performs the Wald-Wolfowitz (or Runs) test of the null hypothesis that the given data is random, or independently sampled.
+The data can come as many-valued or two-valued (boolean). If many-valued, the sample is transfromed by labelling each
+element as above or below the median.
+
+Implements: [`pvalue`](@ref)
+"""
+function WaldWolfowitzTest(x::AbstractVector{T}) where T<:Bool
```

Comment:
```suggestion
function WaldWolfowitzTest(x::AbstractVector{Bool})
```

Reply:
In my testing, the array `x .>=med` is of type `bitmask` so just saying `Bool` did not work.

Reply:
Not it's a `BitArray` but that's OK. `Bool` is a concrete type anyway, it doesn't have subtypes.